sympy/physics/mechanics/kane.py

```diff
@@ -330,15 +360,64 @@ def _initialize_constraint_matrices(self, config, vel, acc, linear_solver='LU'):
             self._k_dnh = Matrix()
             self._Ars = Matrix()
 
+        # Initialize nonlinear velocity constraints
+        # The idea is this:
+        # 0 = nonlin_vel.diff(t) is linear in udot, just like vel is linear in u.
+        if nonlin_vel:
+
+            # When calling kanes_equations, another class instance will be
+            # computation of kinetic differential equation matrices will be
+            # skipped as this was computed during the original KanesMethod
+            # object, and the qd_u_map will not be available.
+            if self._qdot_u_map is not None:
+                nonlin_vel = msubs(nonlin_vel, self._qdot_u_map)
+            nonlin_veldt = nonlin_vel.diff(dynamicsymbols._t)
+            self._k_c, self._f_c_neg = linear_eq_to_matrix(nonlin_veldt, self._udot[:])
+            self._f_c = -self._f_c_neg
```

Comment:
```suggestion
            self._k_c, f_c_neg = linear_eq_to_matrix(nonlin_veldt, self._udot[:])
            self._f_c = -f_c_neg
```
No good reason to store the negative as class attribute.

Reply:
> No good reason to store the negative as class attribute.

Makes sense! I will change this.
Question: Is there a draw back in storing unneeded things, or "just" bad style?

Reply:
The purpose of storing an variable on a class is so it can be used in multiple methods. If you only use it in one method, no reason to store it (unless you want the user to have easy access to the variable).

Reply:
The draw backs are those that are coupled to bad style. It is redundancy, cluttering of the attributes, resulting in worse autocompletion/readability, etc. Also, people may actually start using both of these attributes and then you cannot simply remove it anymore.

Reply:
> The draw backs are those that are coupled to bad style. It is redundancy, cluttering of the attributes, resulting in worse autocompletion/readability, etc. Also, people may actually start using both of these attributes and then you cannot simply remove it anymore.

I thought variables whose name begins with un understroke should not be used by the user - or at least at his own risk?
All else I understand.

Reply:
They shouldn't, but it does happen.